src/components/PhoneInput/hooks/useHandleFocus.js

```diff
@@ -0,0 +1,16 @@
+import { useCallback } from 'react';
+
+export default function useHandleFocus(focusIndex, onFocus, setFocusIndex) {
+    return useCallback(
+        (event, index) => {
+            if ((index === 0 && focusIndex === 1) || index === focusIndex) {
+                return;
+            }
+            if (focusIndex === -1) {
+                onFocus(event);
+            }
+            setFocusIndex(index);
```

Comment:
leave this for now, but keep in mind that passing setState to external functions is an antipattern